### PR TITLE
APIs to hide/unhide chat for a user

### PR DESCRIPTION
**Pull request by RamjotSingh**

### CODE REVIEW

File: api-reference/beta/api/chat-hideforuser.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
+---
+title: "chat: hideForUser"
+description: "Hide chat for a user"
```
Comment (JarbasHorst) [inline api-reference/beta/api/chat-hideforuser.md:3]: Articles help readers and translation software identify the nouns and modifiers in a sentence. For clarity, please add a before chat. Suggested revision:

- Hide a chat for a user.

File: api-reference/beta/api/chat-hideforuser.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+ms.prod: "microsoft-teams"
+doc_type: apiPageType
+---
+
+# chat: hideForUser
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Hides [chat](../resources/chat.md) for a user.
```
Comment (JarbasHorst) [inline api-reference/beta/api/chat-hideforuser.md:15]: Articles help readers and translation software identify the nouns and modifiers in a sentence. For clarity, please add a before chat. Suggested revision:

- Hides a [chat]...

File: api-reference/beta/api/chat-hideforuser.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+> **Note:** Chat will be automatically unhidden for the user if an action (like sending a message) is taken on the chat.
```
Comment (JarbasHorst) [inline api-reference/beta/api/chat-hideforuser.md:17]: For clarity, I recommend you to rephrase this sentence. Suggested revision:

- A chat is automatically unhidden for a user if an action such as *Send Message* is taken at chat level.

File: api-reference/beta/api/chat-hideforuser.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+```
+
+## Request headers
+|Name|Description|
+|:---|:---|
+|Authorization|Bearer {token}. Required.|
+|Content-Type|application/json. Required.|
+
+## Request body
+In the request body, supply JSON representation of the parameters.
```
Comment (JarbasHorst) [inline api-reference/beta/api/chat-hideforuser.md:45]: Please, add a before JSON. Suggested revision:

- ...supply a JSON...

File: api-reference/beta/api/chat-hideforuser.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+The following table shows the parameters that can be used with this action.
+
+|Parameter|Type|Description|
+|:---|:---|:---|
+|user|[teamworkUserIdentity](../resources/teamworkuseridentity.md)|User to hide the chat for. **In delegated, a user may only hide chat for themselves**.|
```
Comment (JarbasHorst) [inline api-reference/beta/api/chat-hideforuser.md:51]: For conformity and clarity, please 

- add mode after delegated
- replace a user with users
- add a before chat

 Suggested revision:

- In delegated mode, users may only hide a chat for themselves.

File: api-reference/beta/api/chat-hideforuser.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+|tenantId|String|Tenant id of the user.|
```
Comment (JarbasHorst) [inline api-reference/beta/api/chat-hideforuser.md:52]: In regular text, please use ID as the abbreviation for identity.

File: api-reference/beta/api/chat-unhideforuser.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
+---
+title: "chat: unhideForUser"
+description: "Unhide chat for a user"
```
Comment (JarbasHorst) [inline api-reference/beta/api/chat-unhideforuser.md:3]: Please, add a period at the end of the sentence.

Comment (JarbasHorst) [inline api-reference/beta/api/chat-unhideforuser.md:3]: Articles help readers and translation software identify the nouns and modifiers in a sentence. For clarity, please add a before chat. Suggested revision:

- Unhide a chat for a user.

File: api-reference/beta/api/chat-unhideforuser.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+ms.prod: "microsoft-teams"
+doc_type: apiPageType
+---
+
+# chat: unhideForUser
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Unhides [chat](../resources/chat.md) for a user.
```
Comment (JarbasHorst) [inline api-reference/beta/api/chat-unhideforuser.md:15]: Articles help readers and translation software identify the nouns and modifiers in a sentence. For clarity, please add a before chat. Suggested revision:

- Unhide a chat for a user.

File: api-reference/beta/api/chat-unhideforuser.md [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+|Content-Type|application/json. Required.|
+
+## Request body
+In the request body, supply JSON representation of the parameters.
+
+The following table shows the parameters that can be used with this action.
+
+|Parameter|Type|Description|
+|:---|:---|:---|
+|user|[teamworkUserIdentity](../resources/teamworkuseridentity.md)|User to unhide the chat for. **In delegated, a user may only unhide chat for themselves**.|
```
Comment (JarbasHorst) [inline api-reference/beta/api/chat-unhideforuser.md:49]: For conformity and clarity, please 

- add mode after delegated
- replace a user with users
- add a before chat

 Suggested revision:

- In delegated mode, users may only unhide a chat for themselves.

File: api-reference/beta/api/chat-unhideforuser.md [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+|tenantId|String|Tenant id of the user.|
```
Comment (JarbasHorst) [inline api-reference/beta/api/chat-unhideforuser.md:50]: In regular text, please use ID as the abbreviation for identity.

File: api-reference/beta/api/chat-unhideforuser.md [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+```
+
+## Request headers
+|Name|Description|
+|:---|:---|
+|Authorization|Bearer {token}. Required.|
+|Content-Type|application/json. Required.|
+
+## Request body
+In the request body, supply JSON representation of the parameters.
```
Comment (JarbasHorst) [inline api-reference/beta/api/chat-unhideforuser.md:43]: Please, add a before JSON. Suggested revision:

- ...supply a JSON...

File: api-reference/beta/resources/chat.md [hunk 12]
```diff
@@ -31,6 +31,8 @@ A chat is a collection of [chatMessages](chatmessage.md) between one or more par
 |[Get chat member](../api/chat-get-members.md) | [conversationMember](conversationmember.md) | Get a single user in the chat.| 
 |[Remove chat member](../api/chat-delete-members.md)|None|Remove a user from the chat.|
 |[Get chat between user and app](../api/userscopeteamsappinstallation-get-chat.md) | [chat](chat.md)| Get one-on-one chat between user and the app |
+|[hideForUser](../api/chat-hideforuser.md)|None|Hide chat for a user.|
```
Comment (JarbasHorst) [inline api-reference/beta/resources/chat.md:34]: For clarity and consistency, please add a before chat. Suggested revision:

- Hide a chat..

File: api-reference/beta/resources/chat.md [hunk 13]
```diff
@@ -31,6 +31,8 @@ A chat is a collection of [chatMessages](chatmessage.md) between one or more par
...
+|[unhideForUser](../api/chat-unhideforuser.md)|None|Unhide chat for a user.|
```
Comment (JarbasHorst) [inline api-reference/beta/resources/chat.md:35]: For clarity and consistency, please add a before chat. Suggested revision:

- Unhide a chat..

**Review by JarbasHorst — CHANGES_REQUESTED** [review]

File: api-reference/v1.0/api/chat-post-messages.md [hunk 14]
```diff
@@ -11,7 +11,7 @@ doc_type: "apiPageType"
 
 Namespace: microsoft.graph
 
-Send a new [chatMessage](../resources/chatmessage.md) in the specified [chat](../resources/chat.md). This API cannot create a new chat; you must use the [list chats](chat-list-messages.md) method to retrieve the ID of an existing chat before creating a chat message.
+Send a new [chatMessage](../resources/chatmessage.md) in the specified [chat](../resources/chat.md). This API cannot create a new chat; you must use the [list chats](chat-list.md) method to retrieve the ID of an existing chat before creating a chat message.
```
Comment (JarbasHorst) [inline api-reference/v1.0/api/chat-post-messages.md:14]: Please, change the sentence to conform with the Microsoft style. Suggested revision:

- This API can't create a new chat; ...chat before you can create a chat message.

File: api-reference/v1.0/api/chat-post-messages.md [hunk 15]
```diff
@@ -11,7 +11,7 @@ doc_type: "apiPageType"
...
 
 > **Note**: We don't recommend that you use this API for data migration. It does not have the throughput necessary for a typical migration.
```
Comment (JarbasHorst) [inline api-reference/v1.0/api/chat-post-messages.md:16]: Please, use common contractions in your articles, such as doesn't, to create a friendly, informal tone. In this case, please replace does not with doesn't

File: changelog/Microsoft.Teams.Core.json [hunk 16]
```diff
@@ -1,6 +1,40 @@
...
+          "Id": "6adc71c8-6be6-473f-96ea-d00d71babea6",
+          "ApiChange": "Property",
+          "ChangedApiName": "isHidden",
+          "ChangeType": "Addition",
+          "Description": "Added the **isHidden** property to [chatViewpoint](https://docs.microsoft.com/en-us/graph/api/resources/chatViewpoint?view=graph-rest-beta) resource.",
+          "Target": "chatViewpoint"
+        },
+        {
+          "Id": "6adc71c8-6be6-473f-96ea-d00d71babea6",
+          "ApiChange": "Action",
```
Comment (JarbasHorst) [inline changelog/Microsoft.Teams.Core.json:16]: Action isn't on the list of approved values for ApiChange. Please change this to "Method".

File: changelog/Microsoft.Teams.Core.json [hunk 17]
```diff
@@ -1,6 +1,40 @@
...
+          "ChangedApiName": "hideForUser",
+          "ChangeType": "Addition",
+          "Description": "Added the [hideForUser](https://docs.microsoft.com/en-us/graph/api/hideForUser?view=graph-rest-beta) method to the [hideForUser](https://docs.microsoft.com/en-us/graph/api/hideForUser?view=graph-rest-beta) resource.",
+          "Target": "hideForUser"
+        },
+        {
+          "Id": "6adc71c8-6be6-473f-96ea-d00d71babea6",
+          "ApiChange": "Action",
```
Comment (JarbasHorst) [inline changelog/Microsoft.Teams.Core.json:24]: Action isn't on the list of approved values for ApiChange. Please change this to "Method".

File: changelog/Microsoft.Teams.Core.json [hunk 18]
```diff
@@ -1,6 +1,40 @@
...
 
   "changelog": [
+    {
+      "ChangeList": [
+        {
+          "Id": "6adc71c8-6be6-473f-96ea-d00d71babea6",
+          "ApiChange": "Property",
+          "ChangedApiName": "isHidden",
+          "ChangeType": "Addition",
+          "Description": "Added the **isHidden** property to [chatViewpoint](https://docs.microsoft.com/en-us/graph/api/resources/chatViewpoint?view=graph-rest-beta) resource.",
```
Comment (JarbasHorst) [inline changelog/Microsoft.Teams.Core.json:11]: Articles help readers and translation software identify the nouns and modifiers in a sentence. Please, add the before chatViewpoint. Suggested revision:

- ...property to the [chatViewpoint]...

File: changelog/Microsoft.Teams.Core.json [hunk 19]
```diff
@@ -1,6 +1,40 @@
...
+          "ChangeType": "Addition",
+          "Description": "Added the [hideForUser](https://docs.microsoft.com/en-us/graph/api/hideForUser?view=graph-rest-beta) method to the [hideForUser](https://docs.microsoft.com/en-us/graph/api/hideForUser?view=graph-rest-beta) resource.",
+          "Target": "hideForUser"
+        },
+        {
+          "Id": "6adc71c8-6be6-473f-96ea-d00d71babea6",
+          "ApiChange": "Action",
+          "ChangedApiName": "unhideForUser",
+          "ChangeType": "Addition",
+          "Description": "Added the [unhideForUser](https://docs.microsoft.com/en-us/graph/api/unhideForUser?view=graph-rest-beta) method to the [unhideForUser](https://docs.microsoft.com/en-us/graph/api/unhideForUser?view=graph-rest-beta) resource.",
```
Comment (JarbasHorst) [inline changelog/Microsoft.Teams.Core.json:27]: Please, replace "[unhideForUser](https://docs.microsoft.com/en-us/graph/api/unhideForUser?view=graph-rest-beta) resource" with the correct resource.

File: changelog/Microsoft.Teams.Core.json [hunk 20]
```diff
@@ -1,6 +1,40 @@
...
+          "ChangeType": "Addition",
+          "Description": "Added the **isHidden** property to [chatViewpoint](https://docs.microsoft.com/en-us/graph/api/resources/chatViewpoint?view=graph-rest-beta) resource.",
+          "Target": "chatViewpoint"
+        },
+        {
+          "Id": "6adc71c8-6be6-473f-96ea-d00d71babea6",
+          "ApiChange": "Action",
+          "ChangedApiName": "hideForUser",
+          "ChangeType": "Addition",
+          "Description": "Added the [hideForUser](https://docs.microsoft.com/en-us/graph/api/hideForUser?view=graph-rest-beta) method to the [hideForUser](https://docs.microsoft.com/en-us/graph/api/hideForUser?view=graph-rest-beta) resource.",
```
Comment (JarbasHorst) [inline changelog/Microsoft.Teams.Core.json:19]: Please, replace "[hideForUser](https://docs.microsoft.com/en-us/graph/api/hideForUser?view=graph-rest-beta) resource" with the correct resource.

**Review by JarbasHorst — CHANGES_REQUESTED** [review]

**Comment by JarbasHorst** [comment]
@RamjotSingh, checking to see if you have any questions regarding some of the feedback I provided?

**Comment by JarbasHorst** [comment]
Hi @RamjotSingh, I just dropped by to see if you have any questions regarding some of the feedback I provided?

File: api-reference/beta/api/chat-hideforuser.md [hunk 21]
```diff
@@ -0,0 +1,89 @@
...
+doc_type: apiPageType
+ms.localizationpriority: medium
+---
+
+# chat: hideForUser
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Hides a [chat](../resources/chat.md) for a user.
```
Comment (JarbasHorst) [inline api-reference/beta/api/chat-hideforuser.md:15]: The introductory sentence for API topics should start with an imperative verb. Suggested revision:

- Hide a [chat]...

File: api-reference/beta/api/chat-unhideforuser.md [hunk 22]
```diff
@@ -0,0 +1,87 @@
...
+doc_type: apiPageType
+ms.localizationpriority: medium
+---
+
+# chat: unhideForUser
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Unhides a [chat](../resources/chat.md) for a user.
```
Comment (JarbasHorst) [inline api-reference/beta/api/chat-unhideforuser.md:15]: The introductory sentence for API topics should start with an imperative verb. Suggested revision:

- Unhide a [chat]...

File: api-reference/beta/api/chatmessage-post.md [hunk 23]
```diff
@@ -28,10 +28,12 @@ One of the following permissions is required to call this API. To learn more, in
 |:---------------------------------------|:--------------------------------------------|
 | Delegated (work or school account)     | ChannelMessage.Send, Group.ReadWrite.All** |
 | Delegated (personal Microsoft account) | Not supported. |
-| Application                            | Not supported. |
+| Application                            | Teamwork.Migrate.All. |
```
Comment (JarbasHorst) [inline api-reference/beta/api/chatmessage-post.md:31]: For consistency, please remove the period after Teamwork.Migrate.All. Suggested revision:

- | Application                            | Teamwork.Migrate.All |

File: api-reference/beta/resources/chatviewpoint.md [hunk 24]
```diff
@@ -20,6 +20,7 @@ Represents user-specific properties of a [chat](../resources/chat.md). These pro
 ## Properties
 |Property|Type|Description|
 |:---|:---|:---|
+|isHidden|Boolean|Represents if the chat is hidden for the user.|
```
Comment (JarbasHorst) [inline api-reference/beta/resources/chatviewpoint.md:23]: Hi @RamjotSingh,

according to Anand, the description should be consistency. Based on his suggestion, you should either use:

- (*my favorite*) Represents if the chat is hidden for the current user.
and update the text for the next item to also have the term current user.

OR

- Represents if the chat is hidden for the calling user.

File: api-reference/v1.0/api/chatmessage-post.md [hunk 25]
```diff
@@ -26,7 +26,9 @@ One of the following permissions is required to call this API. To learn more, in
 |:---------------------------------------|:--------------------------------------------|
 | Delegated (work or school account)     | ChannelMessage.Send |
 | Delegated (personal Microsoft account) | Not supported. |
-| Application                            | Not supported. |
+| Application                            | Teamwork.Migrate.All. |
```
Comment (JarbasHorst) [inline api-reference/v1.0/api/chatmessage-post.md:29]: For consistency, please remove the period after Teamwork.Migrate.All. Suggested revision:

- | Application                            | Teamwork.Migrate.All |

**Comment by JarbasHorst** [comment]
@RamjotSingh, please include an update to the toc.yml file in the beta folder, to surface these two new APIs in the TOC.

File: api-reference/beta/resources/chat.md [hunk 26]
```diff
@@ -33,6 +33,8 @@ A chat is a collection of [chatMessages](chatmessage.md) between one or more par
 |[Get chat between user and app](../api/userscopeteamsappinstallation-get-chat.md) | [chat](chat.md)| Get one-on-one chat between user and the app.|
 |[Mark chat as read](../api/chat-markchatreadforuser.md) |None| Mark chat as read for a user.|
 |[Mark chat as unread](../api/chat-markchatunreadforuser.md) |None| Mark chat as unread for a user.|
+|[Hide a chat for user](../api/chat-hideforuser.md)|None|Hide a chat for a user.|
```
Comment (JarbasHorst) [inline api-reference/beta/resources/chat.md:36]: For consistency with other entries in this table, I recommend you to update the values in the Method column to not contain "for user" and the article. Suggested revision:

- Hide chat
- Unhide chat

I recommend you to also use these values in the TOC.

**Review by JarbasHorst — CHANGES_REQUESTED** [review]

**Comment by JarbasHorst** [comment]
@Lauragra this PR is ready for final review.

**Review by Lauragra — APPROVED** [review]

**Pull request merged**